packages/dds/merge-tree/Obliterate.md

```diff
@@ -0,0 +1,276 @@
+For example, `{ seq: 55, clientId: 5, refSeq: 50, op: <insert "i won't be obliterated" at index 10> }`.
+
+The forward excursion would need to continue past this segment in order to conclude it isn't in an obliterated range.
+If that was the only such concurrent insert, the next segment it would visit would be an obliterated one and we'd decide
+whether or not to include the newly inserted segment as part of the obliterated region based on some endpoint merge strategy.
+
+The key insight is that visiting the segment with seq 55 does provide the excursion with information: since the segment
+was inserted at seq 55 and isn't obliterated, any obliterate operation must have occurred before seq 55.
+If we keep track of the smallest sequence number that we've visited, we can halt the excursion as soon as it falls below
+the smallest obliterate operation within the collab window.
+If we alternatively reach a segment that has been obliterated concurrently to the insert we're processing, we can also stop
+and use the endpoint resolution strategy.
+
+All-in-all, the insert logic modification might look something like this:
+
+```typescript
+function insertingWalk(args /* mostly omitted */, op) {
+  /* regular insert logic goes here */
+
+  let currentMin = Number.POSITIVE_INFINITY;
+  let obliteratedSegment: ISegment | undefined = undefined;
+  const smallestSeqObliterateOp = this.getSmallestSeqObliterateOp();
+  const findAdjacentObliteratedSegment = (seg) => {
+    if (seg.seq === UnassignedSequenceNumber) {
+      // Ignore un-acked segments
+      return true;
+    }
+
+    if (seg.obliteratedSeq && seg.obliteratedSeq > op.referenceSequenceNumber) {
+      obliteratedSegment = seg;
+      return false;
+    }
+
+    currentMin = Math.min(currentMin, seg.seq);
+    // If we've reached a segment that existed before any of our in-collab-window obliterate ops
+    // happened, no need to continue.
+    return currentMin > smallestSeqObliterateOp;
+  }
+  forwardExcursion(insertSegment, findAdjacentObliteratedSegment);
+  const furtherObliteratedSegment = obliteratedSegment;
+  currentMin = Number.POSITIVE_INFINITY;
+  obliteratedSeg = undefined;
+  backwardExcursion(insertSegment, findAdjacentObliteratedSegment);
+  const nearerObliteratedSegment = obliteratedSegment;
+  if (
+    (nearerObliteratedSegment && breakEndpointTie(nearerObliteratedSegment, insertSegment, op)) ||
+    (furtherObliteratedSegment && breakEndpointTie(insertSegment, furtherObliteratedSegment, op))
+  ) {
+    // These objects will be analogous to return from `toRemovalInfo`.
+    const nearObliterateInfo = toObliterateInfo(nearerObliteratedSegment);
+    const farObliterateInfo = toObliterateInfo(furtherObliteratedSegment);
+    // The inserted segment could potentially be adjacent to two different obliterated regions.
+    // We mark it as obliterated using the info from the earlier such operation.
+    const obliteratingInfo = min(nearObliterateInfo, farObliterateInfo);
```

Comment:
i think we might want the max here. the obliterate seq is like a waterline that need to rise to capture incoming changes that fall under the waterline